llvm/lib/Transforms/Scalar/SROA.cpp

```diff
@@ -2304,6 +2305,34 @@ static VectorType *isVectorPromotionViable(Partition &P, const DataLayout &DL) {
       }
     }
   };
+  auto createAndCheckVectorTypesForPromotion =
```

Comment:
Can this be a regular helper function? I don't see what context this is capturing

Reply:
Thanks for comment.

I originally had it as a lambda because it calls the lambda CheckCandidateType which captures the flags / heuristics (e.g. HaveCommonVecPtrTy)  (among other things).

The helper method is probably fine, but I find the scope behavior a little weird.